src/l500/l500-options.cpp

```diff
@@ -63,28 +85,113 @@ namespace librealsense
         auto max_value = float(*(reinterpret_cast<int32_t*>(max.data())));
         auto min_value = float(*(reinterpret_cast<int32_t*>(min.data())));
 
+        bool success;
+        auto res = query_default( success );
+
+        if( !success )
+        {
+            _is_read_only = true;
+            res = -1;
+        }
+
         _range = option_range{ min_value,
-            max_value,
-            float(*(reinterpret_cast<int32_t*>(step.data()))),
-            def };
+                               max_value,
+                               float( *( reinterpret_cast< int32_t * >( step.data() ) ) ),
+                               res };
+    }
+
+    void l500_hw_options::set_default( float def )
+    {
+        _range.def = def;
+    }
+
+    void l500_hw_options::set_read_only( bool read_only )
+    {
+        _is_read_only = read_only;
+    }
+
+    void l500_hw_options::set_manually( bool set ) 
+    {
+        _was_set_manually = set;
     }
 
     void l500_hw_options::enable_recording(std::function<void(const option&)> recording_action)
     {}
 
-    float l500_hw_options::query(int mode) const
+    // this mathod can throw an exception if un-expected error occurred
+    // or return error by output parameter 'success' if default value not applicable
+    // on this state we return -1 as default value
```

Comment:
I don't think the user should expect -1. Make them count on `success`, not the -1. If success is `false`, return value is undefined.

Reply:
fixed the comment